### ISSUE TITLE
Non-Mock Requests Missing Headers

### ISSUE THREAD

**Issue by ookadoo** [nutboltu/storybook-addon-mock#71]
I have stories that do not use this addon which stopped working after I installed it.  It appears the headers are being stripped when going to the destination.  These are headless CMS requests and I don't want to mock them.

If someone can point me to the passthrough portion of the code, I'd be glad to try and submit a PR.

**Comment by nutboltu** [comment]
@ookadoo would you please provide a reproducing code snippet so that I can check what's the actual issue?

**Comment by charlesgivens** [comment]
There may be some validity to this. When I don't have mocks implemented, everything works fine. As soon As I mock a request, a bunch of unrelated requests start to fail. They never make to my network tab so I can't give any data.

**Comment by charlesgivens** [comment]
A way to test this is to create a promise that makes 2 separate api requests on inside the other(This is not uncommon with api that perform a separate Auth api calls. Only mock one of them in your story. This should give the same bug.

**Comment by fantun3s** [comment]
I am having a similar issue but in my case, both the headers and the body of requests performed on stories that don't use the storybook-addon-mock are getting removed

**Comment by synchronos-t** [comment]
I seem to have a similar issue. If I just import storybook-addon-mock, another request not using it fails, because it stops making the CORS preflight with OPTIONS, and then for some reason it makes the actual POST with incorrect Accept header, which makes the backend respond with 415. When I remove the storybook-addon-mock import, it starts working again.

**Comment by Bangybug** [comment]
I have the same issue with storybook-addon-mock. I traced it down to storybook-addon-mock/src/utils/faker.js

See how it handles unmatched requests in mockXhrRequest method:
```
mockXhrRequest = (xhr) => {
  if (matched) {
  ...
  } else {
  ...
   const realXhr = new global.realXMLHttpRequest();
   realXhr.open(method, url); 
   ...
   realXhr.send(body);
   ...
   }
}
```

As you can see, the headers are not being sent with realXhr. Headers are accessible via xhr.requestHeaders, so it should be really simple fix.

**Comment by neil-morrison44** [comment]
I think this issue also breaks viewing Google Maps Vector maps within storybook when the plugin is being used

**Comment by nutboltu** [comment]
The issue is fixed in [2.4.1](https://github.com/nutboltu/storybook-addon-mock/releases/tag/2.4.1)
